src/DDTrace/Integrations/Laravel/LaravelSandboxedIntegration.php

```diff
@@ -0,0 +1,206 @@
+    }
+
+    /**
+     * @return int
+     */
+    public function init()
+    {
+        if (!Configuration::get()->isIntegrationEnabled(LaravelSandboxedIntegration::NAME)) {
+            return SandboxedIntegration::NOT_LOADED;
+        }
+
+        $rootScope = GlobalTracer::get()->getRootScope();
+        $rootSpan = null;
+
+        if (null === $rootScope || null === ($rootSpan = $rootScope->getSpan())) {
+            return SandboxedIntegration::NOT_LOADED;
+        }
+
+        $integration = $this;
+
+        \dd_trace_method(
+            'Illuminate\Foundation\Application',
+            'handle',
+            function (SpanData $span, $args, $response) use ($rootSpan, $integration) {
+                // Overwriting the default web integration
+                $rootSpan->overwriteOperationName('laravel.request');
+                $rootSpan->setIntegration($integration);
+                $rootSpan->setTraceAnalyticsCandidate();
+                if (\method_exists($response, 'getStatusCode')) {
+                    $rootSpan->setTag(Tag::HTTP_STATUS_CODE, $response->getStatusCode());
+                }
+                $rootSpan->setTag(Tag::SERVICE_NAME, $integration->getServiceName());
+
+                $span->name = 'laravel.application.handle';
+                $span->type = Type::WEB_SERVLET;
+                $span->service = $integration->getServiceName();
+                $span->resource = 'Illuminate\Foundation\Application@handle';
+            }
+        );
+
+        \dd_trace_method(
+            'Illuminate\Routing\Router',
+            'findRoute',
+            function (SpanData $span, $args, $route) use ($rootSpan, $integration) {
+                if (null === $route) {
+                    return false;
+                }
+
+                list($request) = $args;
+
+                // Overwriting the default web integration
+                $rootSpan->setIntegration($integration);
+                $rootSpan->setTraceAnalyticsCandidate();
+                $rootSpan->setTag(
+                    Tag::RESOURCE_NAME,
+                    $route->getActionName() . ' ' . ($route->getName() ?: 'unnamed_route')
+                );
+                $rootSpan->setTag('laravel.route.name', $route->getName());
+                $rootSpan->setTag('laravel.route.action', $route->getActionName());
+                $rootSpan->setTag('http.url', $request->url());
+                $rootSpan->setTag('http.method', $request->method());
+
+                return false;
+            }
+        );
+
+        \dd_trace_method(
+            'Illuminate\Routing\Route',
+            'run',
+            function (SpanData $span) use ($integration) {
+                $span->name = 'laravel.action';
+                $span->type = Type::WEB_SERVLET;
+                $span->service = $integration->getServiceName();
+                $span->resource = $this->uri;
+                $span->meta['integration.name'] = LaravelSandboxedIntegration::NAME;
```

Comment:
The `integration.name` tag is [only available for testing when `DD_TEST_INTEGRATION=1`](https://github.com/DataDog/dd-trace-php/blob/9662c09db74666ad4417f711a9875f788998e793/src/DDTrace/Encoders/SpanEncoder.php#L74-L82). Since internal spans do not have the concept of an integration yet, we might have to treat this in a similar manner as [setting a trace analytics candidate](https://github.com/DataDog/dd-trace-php/blob/9662c09db74666ad4417f711a9875f788998e793/src/DDTrace/Integrations/SandboxedIntegration.php#L17-L23). Or do you have other thoughts on what we could do here? :)

Reply:
very good idea. I like it 👍